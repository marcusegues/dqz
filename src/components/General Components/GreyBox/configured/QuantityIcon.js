// @flow
import React from 'react';
// $FlowFixMe
import { Text, StyleSheet } from 'react-native';
import { GreyBox } from '../GreyBox';

const ownStyles = StyleSheet.create({
  quantityText: {
    color: 'black',
    fontSize: 14,
    fontFamily: 'roboto_regular',
  },
});

type QuantityIconProps = {
  quantity: number | string,
  unit: string,
};

export const QuantityIcon = ({ quantity, unit }: QuantityIconProps) => (
  <GreyBox>
    <Text style={ownStyles.quantityText}>{`${quantity} ${unit}`}</Text>
  </GreyBox>
);
