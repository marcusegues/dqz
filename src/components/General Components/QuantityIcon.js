// @flow
import React from 'react';
// $FlowFixMe
import { View, Text, StyleSheet } from 'react-native';

const ownStyles = StyleSheet.create({
  quantityContainer: {
    paddingHorizontal: 8,
    paddingVertical: 2,
    backgroundColor: '#E0E0E1',
    borderRadius: 3,
  },
  quantityText: {
    color: 'black',
    fontSize: 14,
    fontFamily: 'roboto_regular',
  },
});

type QuantityIconProps = {
  quantity: number,
  unit: string,
};

const QuantityIcon = ({ quantity, unit }: QuantityIconProps) => (
  <View style={ownStyles.quantityContainer}>
    <Text style={ownStyles.quantityText}>{`${quantity} ${unit}`}</Text>
  </View>
);

export default QuantityIcon;
