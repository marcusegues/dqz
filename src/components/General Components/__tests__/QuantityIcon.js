import React from 'react';
import renderer from 'react-test-renderer';
import { QuantityIcon } from '../QuantityIcon';

describe('QuantityIcon', () => {
  test('renders according to snapshot', () => {
    const component = renderer
      .create(<QuantityIcon quantity={100} unit="foos" />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
