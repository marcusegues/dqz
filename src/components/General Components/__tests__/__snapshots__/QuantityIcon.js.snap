// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantityIcon renders according to snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E0E0E1",
      "borderRadius": 3,
      "paddingHorizontal": 8,
      "paddingVertical": 2,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
        "fontFamily": "roboto_regular",
        "fontSize": 14,
      }
    }
  >
    100 foos
  </Text>
</View>
`;
